charts/sonarqube/templates/config.yaml

```diff
@@ -8,18 +8,15 @@ metadata:
     release: {{ .Release.Name }}
     heritage: {{ .Release.Service }}
 data:
-  {{- if and .Values.sonarSecretKey (not .Values.sonarProperties) (not .Values.elasticsearch.bootstrapChecks) }}
-  sonar.properties: sonar.secretKeyPath={{ .Values.sonarqubeFolder }}/secret/sonar-secret.txt
-  {{- end }}
-  {{- if or .Values.sonarProperties (not .Values.elasticsearch.bootstrapChecks) }}
+{{- if or .Values.sonarProperties .Values.sonarSecretKey (not .Values.elasticsearch.bootstrapChecks ) }}
   sonar.properties:
-  {{ range $key, $val := .Values.sonarProperties }}
+  {{- range $key, $val := .Values.sonarProperties }}
     {{ $key }}={{ $val }}
-  {{ end }}
+  {{- end }}
   {{- if not .Values.elasticsearch.bootstrapChecks }}
     sonar.es.bootstrap.checks.disable=true
   {{- end }}
+  {{- if .Values.sonarSecretKey }}
+    sonar.secretKeyPath={{ .Values.sonarqubeFolder }}/secret/sonar-secret.txt
```

Comment:
The path to the encryption key can be specified among the `sonar.properties` (see the [doc](https://docs.sonarqube.org/latest/instance-administration/security/#:~:text=A%20unique%20secret%20key%20must,click%20on%20Generate%20Secret%20Key.&text=Restrict%20file%20permissions%20to%20the,and%20read%2Daccess%20only))).

We should make it sure that the path is not static and gets updated if the `sonar.secretKeyPath` property is set among the sonar.properties. We should probably specify this among the configurable parameters in our helm chart. To sum up: if sonar.secretKeyPath is set among the sonar.properties, we should adjust the path here.

Reply:
That's a good catch, but we will probably not embed that in this PR, as it is a little bit out of the scope of the initial work.